src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java

```diff
@@ -0,0 +1,131 @@
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.maven.doxia.macro.AbstractMacro;
+import org.apache.maven.doxia.macro.Macro;
+import org.apache.maven.doxia.macro.MacroExecutionException;
+import org.apache.maven.doxia.macro.MacroRequest;
+import org.apache.maven.doxia.sink.Sink;
+import org.apache.maven.doxia.sink.impl.SinkEventAttributeSet;
+import org.codehaus.plexus.component.annotations.Component;
+
+/**
+ * A macro that inserts a snippet of code or configuration from a file.
+ */
+@Component(role = Macro.class, hint = "example")
+public class ExampleMacro extends AbstractMacro {
+    /** Starting delimiter for config snippets. */
+    private static final String XML_CONFIG_START = "/*xml";
+
+    /** Ending delimiter for config snippets. */
+    private static final String XML_CONFIG_END = "*/";
+
+    /** Starting delimiter for code snippets. */
+    private static final String CODE_SNIPPET_START = "// xdoc section -- start";
+
+    /** Ending delimiter for code snippets. */
+    private static final String CODE_SNIPPET_END = "// xdoc section -- end";
+
+    /** Newline character. */
+    private static final String NEWLINE = "\n";
+
+    /** Eight whitespace characters. All example source tags are indented 8 spaces. */
+    private static final String EIGHT_SPACES = "        ";
+
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        final String path = (String) request.getParameter("path");
+        final String type = (String) request.getParameter("type");
+        final List<String> lines = readFile("src/xdocs-examples/" + path);
+
+        if ("config".equals(type)) {
+            final String config = getSnippetBetweenDelimiters(lines,
+                    XML_CONFIG_START, XML_CONFIG_END);
+            writeSnippet(sink, config);
+        }
+        else if ("code".equals(type)) {
+            final String code = getSnippetBetweenDelimiters(lines,
+                    CODE_SNIPPET_START, CODE_SNIPPET_END);
+            writeSnippet(sink, code);
+        }
+        else {
+            throw new MacroExecutionException("Unknown example type: " + type);
+        }
+    }
+
+    /**
+     * Read the file at the given path and returns its contents as a list of lines.
+     *
+     * @param path the path to the file to read.
+     * @return the contents of the file as a list of lines.
+     * @throws MacroExecutionException if the file could not be read.
+     */
+    private static List<String> readFile(String path) throws MacroExecutionException {
+        try {
+            final Path exampleFilePath = Path.of(path);
+            return Files.readAllLines(exampleFilePath);
+        }
+        catch (IOException ex) {
```

Comment:
Strange that our checkstyle config allow short name like this, we should use at least  3 chars `exc`.
Please apply to all your changes.

Reply:
The repository has plenty of those
```bash
git grep " catch (IOException ex)"
```
```
src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:        catch (IOException ex) {
src/it/java/com/sun/checkstyle/test/base/AbstractSunModuleTestSupport.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:            catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/ant/CheckstyleAntTask.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoader.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java:            catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java:            catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/gui/MainFrameModel.java:            catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateParser.java:        catch (IOException ex) {
src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java:            catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:        catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheckTest.java:        catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheckTest.java:        catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionFilterTest.java:                catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java:            catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java:            catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java:            catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java:                catch (IOException ex) {
src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java:        catch (IOException ex) {
src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/onestatementperline/InputOneStatementPerLine.java:package com.puppycrawl.tools.checkstyle.checks.coding.onestatementperline; import java.io.IOException; import java.io.OutputStream; import java.io.PipedOutputStream; public class InputOneStatementPerLine {  int e = 1, f = 2, g = 5; private void foo2() { try (OutputStream s12 = new PipedOutputStream(); OutputStream s3 = new PipedOutputStream()) { } catch (IOException ex) { throw new RuntimeException(ex); } } private void foo() { toString(); toString(); } }
src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/redundantmodifier/InputRedundantModifierTryWithResources.java:        } catch (IOException ex) {

```

Reply:
@stoyanK7 yes, `ex` is most common naming aside from `e` in real world usage. Three letter minimum for pretty much any other variable naming is reasonable, I do not see any issue with `ex` for exceptions.

Reply:
Done.

Reply:
https://github.com/checkstyle/checkstyle/issues/13408